### DISCUSSION TITLE
Cursor doesn't change color when switching to insert mode after updating to 23.03

### DISCUSSION THREAD

**Discussion by Nellousan** [helix-editor/helix#10428] [General]
### Summary

Hello,

Since I updated from version 22.12 to 23.03. The cursor color doesn't change anymore when switching from normal mode to insert mode. 

At first i thought maybe it was just a bug with the theme i'm using(everforest-dark) but it seems that every theme has this issue now. (yes i updated the runtime folder as well)

I also believe that the issue does not come from my config file. I tried to open helix as sudo to test(so it doesn't use my config file) and the issue still remains.

Note that i build from source.

### Platform

Linux (EndeavourOs)

### Terminal Emulator

xfce4-terminal 1.0.4

### Helix Version

23.03-31-g1148ce1f

**Comment by the-mikedavis** [comment]
I believe this is from https://github.com/helix-editor/helix/pull/5130. I see the theming for insert/normal/select working on everforest_dark for secondary cursors but not primary cursors. It should fix it to style `ui.cursor.primary.insert` and `ui.cursor.primary.select` as well

**Comment by Nellousan** [comment]
Hello,
good catch, it indeed fixed it.
Is this a bug or should the themes be updated ?

**Comment by the-mikedavis** [comment]
I think that any theme that has `ui.cursor.{insert,select}` scopes and not a `ui.cursor.primary` scope should be updated to add `ui.cursor.primary.{insert,select}`. That should match the behavior before #5130

**Comment by idr4n** [comment]
I'm seeing the same issue in any of my custom themes. I just added `ui.cursor.primary.insert` but I still don't get a different color in insert mode (running helix 23.03 (3cf03723)).
